algolia/api-clients-automation
# chore(javascript): use tsup bundler

## 🧭 What and Why

🎟 JIRA Ticket:

### Changes included:

closes https://github.com/algolia/algoliasearch-client-javascript/issues/1544

use tsup instead of rollup, just because I wanted to try it out and it seems to exactly do what we need, a bit faster and easier to configure.

also add attw and publint, in order to debug the exported types, package.json#export field, and the resolved imports.

- [x] use tsup (chore)
- [x] setup attw and publint (helps with the fix of https://github.com/algolia/algoliasearch-client-javascript/issues/1544)
- [x] update commons (was outdated)
- [x] provide node and browser bundle for clients (same as with rollup, but now it works again)
- [x] provide node bundle for fetch requester (feat)
- [x] use esm mini output from tsup into rollup for umd

## how to verify it works?

- generate and builds the js clients
- go to clients/algoliasearch-client-javascript
- run `test:size`, should be all green without being too far from the threshold
- run `test:bundle` on many clients, they are all green
- try the `5.2.4-beta.2` directly
